src/lib/spinel/spinel.h

```diff
@@ -2955,6 +2963,46 @@ enum
      */
     SPINEL_PROP_THREAD_DOMAIN_NAME = SPINEL_PROP_THREAD_EXT__BEGIN + 44,
 
+    /// Multicast Listeners Register Request
+    /** Format `t(A(6))A(t(CD))` - Write-only
+     * Required capability: `SPINEL_CAP_NET_THREAD_1_2`
+     *
+     * `t(A(6))`: Array of IPv6 multicast addresses
+     * `A(t(CD))`: Array of structs holding optional parameters as follows
```

Comment:
This model of having an array of mapping id to value works fine (sort of like a dictionary) and I think is good here.

The other model I can suggest for optional params (I think we use in some spinel definitions) is to place each param in its own `struct`. When parsing, if struct is empty, we assume it is not provided, otherwise parse the value from the struct.

Reply:
@abtink What you propose has some drawbacks:
- if optional item is not sent the struct is empty, but it's length still needs to be sent, 
- the order of structs means which optional parameter is which (non-empty struct means passev value to some parameter), when there are many such parameters and only last one is provided value all previous need to be sent as zero-length what takes again some bytes to encode the struct length. This is not the case here but it could turn into such as the property evolves. 
However it has advantages:
- it seems simple to parse
- there is possibly such convention already as you say

From your words "... I can suggest ... " I'm not sure if you accept proposed solution or should we change the model for optional parameter as you suggest (Could you please share some sample property where it's used?).

Reply:
@ankuns I think the current model is good here and no need to change. :)

I just suggested it as another way to have in mind for other/new properties.

I agree each model has its pros and cons:
- The dictionary model is good if we have a bunch of optional params. 
   - The downside is we need to define an `id` for each param and then include it in the message (so if we have a lot of optional parameters and most often use many of them, the dictionary model can add the overhead of including the id bytes).
   - We use this model  for dataset (where we have a lot of optional param and a subset can be provided).
- The empty struct `t()` model has the advantage that the spinel format can mirror the ot API call format (when we have pointers to params that can be `null` if not provided). It is more useful when the params are often used. 
  - We use this in few  cases (not that many), e.g. `STREAM` props (for example for PHY data or vendor data), or `ADDRESS_CACHE_TABLE`  or somewhat similar in `MESHCOP_COMMISSIONER_JOINERS` (empty struct to indicate any joiner).